lib/pure/httpcore.nim

```diff
@@ -234,7 +234,7 @@ func parseList(line: string, list: var seq[string], start: int): int =
   while start+i < line.len and line[start + i] notin {'\c', '\l'}:
     i += line.skipWhitespace(start + i)
     i += line.parseUntil(current, {'\c', '\l', ','}, start + i)
-    list.add(current)
+    list.add(move current)
     if start+i < line.len and line[start + i] == ',':
       i.inc # Skip ,
     current.setLen(0)
```

Comment:
Are you sure that isn't pending "near-term/not yet merged" work?  I still get this with HEAD of nim-devel compiled 15 minutes ago:
```
shell$ nim r --hint[Performance]=on --mm:orc tests/stdlib/thttpcore.nim
/usr/lib/nim/lib/pure/httpcore.nim(167, 25) Hint: passing 'headers.table[toCaseInsensitive(headers, key)]' to a sink parameter introduces an implicit copy; if possible, rearrange your program's control flow to prevent it [Performance]
/usr/lib/nim/lib/pure/httpcore.nim(237, 14) Hint: passing 'current' to a sink parameter introduces an implicit copy; if possible, rearrange your program's control flow to prevent it [Performance]
Hint: mm: orc; opt: none (DEBUG BUILD, `-d:release` generates faster code)
47489 lines; 0.641s; 71.609MiB peakmem; proj: thttpcore; out: ....
```
The line 237 one is the relevant warning, of course.  (Not sure how to fix the line 167 one.)  Anyway, happy to remove the line, if it really won't be necessary.

Reply:
Well `move current` does set `current` to "" already and so the `setLen 0` should not be required.